src/fmu/dataio/export/rms/depth_prediction_surfaces.py

```diff
@@ -0,0 +1,112 @@
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Final
+
+import fmu.dataio as dio
+from fmu.dataio._logging import null_logger
+from fmu.dataio._models.fmu_results import product
+from fmu.dataio._models.fmu_results.enums import Classification, ProductName
+from fmu.dataio.export._decorators import experimental
+from fmu.dataio.export._export_result import ExportResult, ExportResultItem
+from fmu.dataio.export.rms._utils import (
+    get_horizons_in_folder,
+    get_rms_project_units,
+    load_global_config,
+)
+
+if TYPE_CHECKING:
+    import xtgeo
+
+_logger: Final = null_logger(__name__)
+
+
+class _ExportDepthPredictionSurfaces:
+    def __init__(
+        self,
+        project: Any,
+        horizon_folder: str,
+    ):
+        _logger.debug("Process data, establish state prior to export.")
+        self._config = load_global_config()
+        self._surfaces = get_horizons_in_folder(project, horizon_folder)
+        self._unit = "m" if get_rms_project_units(project) == "metric" else "ft"
+
+        _logger.debug("Process data... DONE")
+
+    @property
+    def _product(self) -> product.DepthPredictionSurfaceProduct:
+        """Product type for the exported data."""
+        return product.DepthPredictionSurfaceProduct(
+            name=ProductName.depth_prediction_surface
+        )
+
+    @property
+    def _classification(self) -> Classification:
+        """Get default classification."""
+        return Classification.internal
+
+    def _export_surface(self, surf: xtgeo.RegularSurface) -> ExportResultItem:
+        edata = dio.ExportData(
+            config=self._config,
+            content="depth",
+            unit=self._unit,
+            vertical_domain="depth",
+            domain_reference="msl",
+            subfolder="depth_predictions",
+            is_prediction=True,
+            name=surf.name,
+            classification=self._classification,
+            rep_include=False,
+        )
+
+        absolute_export_path = edata._export_with_product(surf, product=self._product)
+        _logger.debug("Surface exported to: %s", absolute_export_path)
+
+        return ExportResultItem(
+            absolute_path=Path(absolute_export_path),
+        )
+
+    def _export_surfaces(self) -> ExportResult:
+        """Do the actual surface export using dataio setup."""
+        return ExportResult(
+            items=[self._export_surface(surf) for surf in self._surfaces]
+        )
+
+    def _validate_surfaces(self) -> None:
+        """Surface validations."""
+        # TODO: Add check that the surfaces are consistent, i.e. a stratigraphic
+        # deeper surface should never have shallower values than the one above
+        # also check that the surfaces have a stratigraphy entry.
+
+    def export(self) -> ExportResult:
+        """Export the depth as a standard_result."""
+        return self._export_surfaces()
+
+
+@experimental
+def export_depth_prediction_surfaces(
```

Comment:
`export_structure_depth_predictions`?
`export_structure_depth_prediction_surfaces`?

Seen in relations to:
`export_structure_thickness_predictions`?

Or just drop the whole "prediction" from the name (implying that this is just "given")?
`export_structure_depth_surfaces`?
`export_structure_thickness_surfaces`?